packages/salesforcedx-vscode-core/src/commands/util/parameterGatherers.ts

```diff
@@ -120,7 +121,15 @@ export class SelectFileName
     CancelResponse | ContinueResponse<{ fileName: string }>
   > {
```

Comment:
Proposed change:
Add a constructor to this class that can take in optional parameters for a `prompt` and a `value`. That way this class can be reused for multiple commands more easily. 
For commands that have multiple user inputs, we can use `CompositeParametersGatherer` which seems to have been written for this purpose. This would fix the issue with commands like `forceFunctionCreate` needing a different prompt value and additional user input.